.test-infra/jenkins/common_job_properties.groovy

```diff
@@ -24,21 +24,33 @@ class common_job_properties {
 
   // Sets common top-level job properties for website repository jobs.
   static void setTopLevelWebsiteJobProperties(context) {
-    setTopLevelJobProperties(context, 'beam-site', 'asf-site', 30)
+    setTopLevelJobProperties(
+            context,
+            'beam-site',
+            'asf-site',
+            'beam',
+            30)
   }
 
   // Sets common top-level job properties for main repository jobs.
   static void setTopLevelMainJobProperties(context,
-                                           String defaultBranch = 'master',
-                                           int defaultTimeout = 100) {
-    setTopLevelJobProperties(context, 'beam', defaultBranch, defaultTimeout)
+                                           String branch = 'master',
+                                           int timeout = 100,
+                                           String label = 'beam') {
+    setTopLevelJobProperties(
+            context,
+            'beam',
+            branch,
+            label,
+            timeout)
   }
 
   // Sets common top-level job properties. Accessed through one of the above
   // methods to protect jobs from internal details of param defaults.
   private static void setTopLevelJobProperties(context,
                                                String repositoryName,
                                                String defaultBranch,
+                                               String defaultLabel,
```

Comment:
defaultLabel -> label (not default, actual)
or, better yet, jenkinsExecutorLabel (ambiguous otherwise)